### DISCUSSION TITLE
GitHub workflow for Scalar.AspNetCore

### DISCUSSION THREAD

**Discussion by xC0dex** [scalar/scalar#3209] [Ideas]
Hey,

I think it would be great to add an automated workflow for the `Scalar.AspNetCore` package. I don't know the best way to start here, so I'll just write down the steps where I think it makes sense to have them.

### General
1.  Always build the .NET solution
When something changes in the package folder (in PRs and in the main branch), `packages/scalar.aspnetcore` build the project to ensure that everything works.

```bash
dotnet build -c Release --no-restore
```

2. Always test the .NET solution
Basically, the same as the first step. After a successful build, it makes sense to test the solution.

```bash
dotnet test -c Release --no-build
```

I think code coverage isn't that important. However, we could add them (at least to the workflow log) if we'd like to.

#### Release

I don't know how the current publish/release workflow is implemented. I always used git tags/releases in the past. In the following example, `$VERSION` is the git tag without the `v`.

```bash
dotnet pack -c Release --no-restore --output nupkgs /p:Version=$VERSION
dotnet nuget push nupkgs/*.nupkg -k ${{ secrets.NUGET_TOKEN }} -s https://api.nuget.org/v3/index.json
```

**Comment by hanspagel** [comment]
Would love to help with this

**Reply by xC0dex** [reply]
That would be great, as I have **no** idea how this changeset works. How would you suggest for a start?

Should we add one new Job to the existing ` .github/workflows/ci.yml` like `build-test-dotnet` file with the steps `build` and `test`?

**Reply by xC0dex** [reply]
I've added a new job in the existing `ci.yml` https://github.com/scalar/scalar/pull/3256. This only builds and tests the .NET package.
